20240328-queries-with-usersets.md

```diff
@@ -0,0 +1,176 @@
+# Meta
+[meta]: #meta
+- Name: Queries with usersets
+- Start Date: 2024-03-28
+- Author(s): @miparnisari
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- Relevant Issues: n/a
+- Supersedes: n/a
+
+# Summary
+[summary]: #summary
+
+This is a proposal to change the outcome of query APIs (specifically, Check and ListObjects APIs) when the target of the query is a userset. The proposal also involves adapting ListUsers API to return responses consistent with Check and ListObjects.
+
+# Definitions
+[definitions]: #definitions
+
+- **Object**: is of the form `objectType:objectId`.
+- **User**: is a specific object, such as `employee:maria` or `document:1`.
+- **Userset**: is a set of objects. Usersets are represented via this notation: `objectType:objectId#relation`. For example, `document:1#viewer` represents the set of objects that are related to `document:1` as `viewer`. There is a special userset of form `objectType:*` that denotes all objects of the given `objectType`.
```

Comment:
`objectType:*`, is not a userset, they're two distinct criteria

Reply:
in the context of set theory, `object:*` is a userset (set of users), no?